### PR TITLE
Fixes #3234: segarray with empty segments and nans parquet bug

### LINKED ISSUE #3234: segarray with empty segments and `nan`s
The fix from https://github.com/Bears-R-Us/arkouda/pull/3233 doesn't seem to handle empty segments and `nan`s (despite this working locally). I'll dig into this more but want the fix for ints to be merged asap since it was a user request

### PR DESCRIPTION
**Pull request by stress-tess**
Fixes #3234 

We found some odd behavior with the existing implementation, so I've opted not to squash my commits in order to make it easier to reproduce in case we want to dig into it more. With a set seed running the same code and versions, it seemed to consistently pass on arm64 macs and fail on x86 linux machines. It not immediately obvious what about the implementation was causing this seemingly architecture specific failure

To sidestep this, I opted for simpler single batch read logic for float/double segarrays. This will be slower but obviously correctness is most important. This is the latest in a string of read logic that I've changed to be single batched, so it might be worth revisiting in the future to see if we can reenable batch reads

### CODE REVIEW

File: src/ArrowFunctions.cpp [hunk 1]
```diff
@@ -644,85 +644,57 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
...
-                chpl_ptr[i+val_idx] = NAN;
-                tmp_offset++; // adjustment for values array since Nulls are not included
-              }
-              else if (def_lvl[j] == 3){ // defined value to write
-                chpl_ptr[i+val_idx] = (double)tmpArr[val_idx-tmp_offset];
+          while (reader->HasNext() && arrayIdx < numElems) {
+            (void)reader->ReadBatch(1, &definition_level, nullptr, &chpl_ptr[arrayIdx], &values_read);
+            // if values_read is 0, that means that it was a null value or empty seg
+            if (values_read != 0) {
+              arrayIdx++;
```
Comment (ajpotts) [inline src/ArrowFunctions.cpp:688]: Why is this different than the `ARROWFLOAT` version?  It's missing this line: `chpl_ptr[arrayIdx] = (double) tmp;`

**Review by ajpotts — APPROVED** [review]
Other than the one comment, it looks OK to me!  I'm not familiar with the Arrow API though :)

File: src/ArrowFunctions.cpp [hunk 2]
```diff
@@ -644,85 +644,57 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
...
-              
-              // Null values treated as NaN
-              if (def_lvl[j] == 2) {
-                chpl_ptr[i+val_idx] = NAN;
-                tmp_offset++; // adjustment for values array since Nulls are not included
-              }
-              else if (def_lvl[j] == 3){ // defined value to write
-                chpl_ptr[i+val_idx] = (double)tmpArr[val_idx-tmp_offset];
+          while (reader->HasNext() && arrayIdx < numElems) {
+            (void)reader->ReadBatch(1, &definition_level, nullptr, &chpl_ptr[arrayIdx], &values_read);
```
Comment (bmcdonald3) [inline src/ArrowFunctions.cpp:685]: I am crying every time I see one of these batch sizes get reduced to 0. I think I know a way around that so we can get the better performance while still checking for nulls (it's similar to my string PR up right now).

We should queue that up for a discussion after ChapelCon gets over.

**Review by bmcdonald3 — APPROVED** [review]
Looks good, unfortunate we are reducing these batch sizes, we can try to fix that in the future.

File: src/ArrowFunctions.cpp (hunk 1 above)
Comment (stress-tess) [reply src/ArrowFunctions.cpp:688]: yah so float is like the 32 bit version of real and double is the 64 bit version.  So we turn the floats into doubles. You can see something similar done to convert the `(u)int32`s into `(u)int64`s

File: src/ArrowFunctions.cpp (hunk 2 above)
Comment (stress-tess) [reply src/ArrowFunctions.cpp:685]: I know 😭

**Review by jaketrookman — APPROVED** [review]

**Pull request merged**